bigml/boostedensemble.php

```diff
@@ -109,7 +109,24 @@ function softmax($predictions) {
         return $predictions;
     }
 
-    function predict($input_data, $by_name=true, $missing_strategy=Tree::LAST_PREDICTION) {
+    function predict($input_data, $by_name=true, $missing_strategy=Tree::LAST_PREDICTION, $compact=null) {
+      /*
+         Makes a prediction based on the prediction made by every model.
+
+         :param input_data: Test data to be used as input
+	     :param by_name: Boolean that is set to true if field_names (as
+	                     alternative to field ids) are used in the
+			             input_data dict
+         :param missing_strategy: numeric key for the individual model's
+                                 prediction method. See the model predict
+                                 method.
+         :param compact: If Null (as is the default), returns the just the prediction. 
+                         If False, prediction is returned as a list of maps, one
+                         per class, with the keys "prediction" and "probability"
+                         mapped to the name of the class and it's probability,
+                         respectively.  If True, returns a list of probabilities
+                         ordered by the sorted order of the class names. 
+      */                    
```

Comment:
There is a mismatch in the definition of $compact as found in boostedensemble.php and ensemble.php -- here, you consider 3 values (true, false, null), with 'null' being the default. On the other hand, ensemble.php only considers true and false and false is the default.